controllers/util/testUtil.go

```diff
@@ -0,0 +1,127 @@
+package util
+
+import (
+	"context"
+
+	"github.com/onsi/ginkgo"
+	ginko "github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	jiraservicedeskv1alpha1 "github.com/stakater/jira-service-desk-operator/api/v1alpha1"
+)
+
+// TestUtil contains necessary objects required to perform operations during tests
+type TestUtil struct {
+	ctx       context.Context
+	k8sClient client.Client
+	r         reconcile.Reconciler
+}
+
+// New creates new TestUtil
+func New(ctx context.Context, k8sClient client.Client, r reconcile.Reconciler) *TestUtil {
+	return &TestUtil{
+		ctx:       ctx,
+		k8sClient: k8sClient,
+		r:         r,
+	}
+}
+
+// CreateNamespace creates a namespace in the kubernetes server
+func (t *TestUtil) CreateNamespace(name string) {
+	namespaceObject := t.CreateNamespaceObject(name)
+	err := t.k8sClient.Create(t.ctx, namespaceObject)
+
+	if err != nil {
+		ginkgo.Fail(err.Error())
+	}
+}
+
+// CreateNamespaceObject creates a namespace object
+func (t *TestUtil) CreateNamespaceObject(name string) *v1.Namespace {
+	return &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+}
+
+// CreateProjectObject creates a jira project custom resource object
+func (t *TestUtil) CreateProjectObject(name string, key string, projectTypeKey string, projectTemplateKey string, description string, assigneeType string, leadAccountId string, url string, namespace string) *jiraservicedeskv1alpha1.Project {
```

Comment:
Input should be a struct, best practice is for a function to have the least possible parameters in it's definition. In this case since these parameters represent the same entity we can group them together in a struct

Reply:
Did this to resolve the circular dependency issue. Using a project object inside testUtil.go initiates a circular dependency issue. Will update it if come up with a better alternative.

Reply:
Resolved.